vllm/v1/core/sched/scheduler.py

```diff
@@ -876,9 +876,26 @@ def update_from_output(
 
         # Remove the stopped requests from the running and waiting queues.
         if stopped_running_reqs:
-            self.running = [
-                req for req in self.running if req not in stopped_running_reqs
-            ]
+            # Optimized removal: single request removal is much faster
+            # than list comprehension for the common case
+            if len(stopped_running_reqs) == 1:
+                # Fast path for single request removal (most common case)
+                stopped_req = next(iter(stopped_running_reqs))
+                if stopped_req in self.running:
```

Comment:
It would be better to just remove and catch `ValueError` to avoid iterating the running queue twice.

Reply:
> It would be better to just remove and catch `ValueError` to avoid iterating the running queue twice.

@njhill Thank you for the excellent feedback! You're absolutely right about avoiding the double traversal. I've updated the implementation to use direct `remove()` with `try-except ValueError` as you suggested